enterprise/reporting/error_index/error_index_reporting.go

```diff
@@ -249,7 +249,7 @@ func (eir *ErrorIndexReporter) mainLoop(ctx context.Context, errIndexDB *jobsdb.
 		"disableSSL":       disableSSL,
 		"enableSSE":        enableSSE,
 	}
-	fm, err := filemanager.NewS3Manager(s3Config, eir.log, func() time.Duration {
+	fm, err := filemanager.NewS3Manager(eir.conf, s3Config, eir.log, func() time.Duration {
```

Comment:
`filemanager.NewS3Manager` should return an interface and pick the proper implementation (v1/v2) based on a `config.Config` option